src/polygon2d.jl

```diff
@@ -0,0 +1,30 @@
+"""
+```
+draw_polygon_perimeter!(img, vertices)
+draw_polygon_perimeter!(img, vertices, color)
+img2=draw_polygon_perimeter(img, vertices)
+img2=draw_polygon_perimeter(img, vertices, color)
+```
+Draws the perimeter of the polygon formed by input vertices(Array of CartesianIndex).
+"""
+
+draw_polygon_perimeter{T<:Colorant}(img::AbstractArray{T, 2}, args...) = draw_polygon_perimeter!(copy(img), args...)
+
+draw_polygon_perimeter!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1})=draw_polygon_perimeter!(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, one(T))
+
+function draw_polygon_perimeter!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, color::T)
+
+    f = CartesianIndex(map(r->first(r)-1, indices(img)))
+    l = CartesianIndex(map(r->last(r), indices(img)))
+    for vertex in vertices
+        if min(f, vertex)!=f || max(l, vertex)!=l
+            println(vertex)
+            error("Polygon coordinates out of range.")
+        end
+    end
```

Comment:
I have updated the function to lines(img, vert, color, closed=true)
In case a point is out-of-bounds, it throws an error after drawing the line segments till that point.